src/components/Accordion.tsx

```diff
@@ -0,0 +1,142 @@
+import { Css, Icon, useTestIds, Palette } from "src/index";
+import { Dispatch, ReactNode, SetStateAction, useEffect, useState } from "react";
+import { useFocusRing } from "react-aria";
+import { useId } from "@react-aria/utils";
+
+export interface AccordionProps {
+  label?: string;
+  title: ReactNode;
+  children: ReactNode;
+  disabled?: boolean;
+  defaultExpanded?: boolean;
+  size?: AccordionSize;
+  /** Adds a top border (enabled by default) */
+  topSection?: boolean;
+  /** Adds a bottom border (disabled by default) */
+  bottomSection?: boolean;
+  /** Used by AccordionList */
+  allowMultipleExpanded?: boolean;
+  index?: number;
+  currentSelectedIndex?: number;
+  setCurrentSelectedIndex?: Dispatch<SetStateAction<number | undefined>>;
+}
+
+type AccordionSize = "xs" | "s" | "m" | "l";
+
+const accordionSizes: Record<AccordionSize, number> = {
+  "xs": 240,
+  "s": 360,
+  "m": 480,
+  "l": 600
+}
+
+export function Accordion(props: AccordionProps) {
+  const {
+    title,
+    children,
+    size,
+    disabled = false,
+    defaultExpanded = false,
+    topSection = true,
+    bottomSection = false,
+    label = "accordion",
+    allowMultipleExpanded = true,
+    index,
+    setCurrentSelectedIndex,
+    currentSelectedIndex
+  } = props;
+  const testIds = useTestIds({}, label);
+  const id = useId();
+  const [expanded, setExpanded] = useState(defaultExpanded);
+  const { isFocusVisible, focusProps } = useFocusRing();
+
+  useEffect(() => {
+    // Handle only one expanded accordion when selected in AccordionList
+    if (!allowMultipleExpanded && index !== currentSelectedIndex) {
+      setExpanded(false)
+    }
+  }, [allowMultipleExpanded, currentSelectedIndex, index]);
+
+  return (
+    <div
+      {...testIds}
+      css={
+       {...Css.mbPx(4)
+        .if(topSection).bt.bGray400
+        .if(bottomSection && !(!expanded && isFocusVisible)).bb.$,
+        ...(size? Css.wPx(accordionSizes[size]).$ : {})} 
+      }
+    >
+      <button
+        {...testIds.summary}
+        {...focusProps}
+        aria-controls={id}
+        aria-expanded={expanded}
+        disabled={disabled}
+        css={Css.w100.df.addIn(":focus", { outline: "none" }).addIn(":hover", Css.bgGray100.$).if(isFocusVisible).bshFocus.$}
+        onClick={() => {
+          setExpanded(!expanded)
+          if (!allowMultipleExpanded && setCurrentSelectedIndex) setCurrentSelectedIndex(index)
+        }}
+      >
+        <div {...testIds.title} css={Css.p2.baseEm.if(disabled).gray500.$}>
+          {title}
+        </div>
+        <div css={Css.ml("auto").my("auto").prPx(14).$}>
```

Comment:
**suggestion:** Can we use flex styles on the button to position the content within?
```
<button css={Css.df.jcsb.gap2.if(disabled).gray500.$}>
  <div>{title}</div>
  <span><Icon /></span>
</button>
```
I'm thinking some of the existing css you have might not be completely necessary. Would be interesting to see how we can achieve the desired styles with less CSS.